test/tools/scripts/extract-testcases.py

```diff
@@ -0,0 +1,227 @@
+        if comment:
+            result += " # " + comment + " #"
+        return result
+
+    def create_isoltest_call(self, line, test_comment):
+        search = re.search(r'ABI_CHECK\(callContractFunction\(\"(.*)\",((.|\s)*)\), encodeArgs\(((.|\s)*)\)\);', line,
+                           re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), self.eval_and_correct(search.group(2)),
+                                               self.eval_and_correct(search.group(4)), test_comment)
+            return result
+
+        search = re.search(r'ABI_CHECK\(callContractFunction\(\"(.*)\",((.|\s)*)\), fromHex\("((.|\s)*)"\)\);', line,
+                           re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), self.eval_and_correct(search.group(2)),
+                                               self.eval_and_correct("0x" + search.group(4)), test_comment)
+            return result
+
+        search = re.search(r'ABI_CHECK\(callContractFunction\("(.*)"\), encodeArgs\(((.|\s)*)\)\);', line, re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), "", self.eval_and_correct(search.group(2)),
+                                               test_comment)
+            return result
+
+        search = re.search(r'ABI_CHECK\(callContractFunction\("(.*)"\), encodeDyn\(((.|\s)*)\)\);', line, re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), "",
+                                               self.eval_and_correct(search.group(2)), test_comment)
+            return result
+
+        search = re.search(r'ABI_CHECK\(callContractFunction\("(.*)"\), fromHex\("((.|\s)*)"\)\);', line, re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), "",
+                                               self.eval_and_correct(search.group(2)), test_comment)
+            return result
+
+        self.extractable = False
+        return "?"
+
+    def extract(self):
+        test_file = open(
+            os.path.dirname(__file__) + "/extracted/libsolidity/semanticTests/end-to-end/" + self.name + ".sol", "w")
+        if self.comment:
+            test_file.write("// " + self.comment.replace("\n", "\n    //"))
+        test_file.write(self.source + "\n")
+
+        if self.alsoViaYul:
+            test_file.write("// ====\n")
+            test_file.write("// compileViaYul: also\n");
+
+        test_file.write("// ----\n")
+        for test in self.tests:
+            test_file.write("// " + test + "\n")
+        test_file.write("\n")
+        test_file.close()
+
+
+def main():
+    test_name = ""
+    test_content = ""
+    inside_test = False
+    tests = {}
+
+    os.makedirs(os.path.dirname(__file__) + "/extracted/libsolidity/semanticTests/end-to-end", 0o777, True)
+    os.makedirs(os.path.dirname(__file__) + "/extracted/libsolidity/ABIJson", 0o777, True)
```

Comment:
What do you need all these for, by the way?

Reply:
`isoltest` is checking for the existence of those directory and will not run if they don't exist. I want to be able to just run this on the generated files with e.g. `./isoltest --no-smt --evmonepath /Users/alex/evmone/lib/libevmone.dylib --testpath /Users/alex/git/aarlt/solidity/test/tools/scripts/extracted`

Reply:
Why not just extract to the repo itself?

Reply:
i just want to keep it separated.. but yeah it may makes sense to have a command line parameter

Reply:
i just want to keep it separated.. but yeah it may makes sense to have a command line parameter